### DISCUSSION TITLE
Automatic lowest refresh rate detection. (Very easy to implement I think)

### DISCUSSION THREAD

**Discussion by AbnerMtz** [seerge/g-helper#2941] [Ideas]
First I would like to thank the devs behind this. You are awesome guys, and single-handedly made a better control interface for our laptops than Asus themselves ❤️

That being said, and diving onto the topic:

I used CRU (Custom Resolution Utility) to lower the refresh rate of my laptop below 60hz (I set it to 48hz, and I disabled the 60hz one) because I want to save even more battery when unplugged. I would like G-helper to **detect** that this is the **lowest refresh rate**. I tried to compile my own version of G-helper, but I couldn't make the code work.

As far as I could tell, G-helper detects the **max refresh rate** automatically. But for the **lowest one**, it doesn't, it just assumes it's 60 (ScreenControl and ScreenNative). Currently when I unplug my laptop it stays in the **highest refresh rate** even when set to automatic. So I must change it manually on the Windows settings. 

**G-helper automatically detecting the lowest refresh rate** would be great and more than enough for me. But ideally, I would like G-helper refresh rates to be more flexible, to **detect all the in-between refresh rates**.

**Comment by seerge** [comment]
@AbnerMtz hello, what is the practical purpose of that ? I don't think there is any real battery saving benefit from setting super low refresh rates via CRU.

And it's actually not that easy to detect :) And in 99.9% of the cases minimum refresh rate would be 60Hz.

**Reply by AbnerMtz** [reply]
thanks for the reply 🙌 I understand perfectly that my case is rare, and I'm sorry I taugh it was easy to detect because i saw the code for detecting the max refresh rate and i taught it would be almost the same but the lowest one. 

That being said, I kindly ask if you could at least tell me in which .cs filles should i search for 60 hz to modify it to 48 for myself and compile my own version? I just figured out ScreenNative and ScreenControl, but i'm not experienced on C# so maybe i'm missing some other file. If i'm asking for too much and you just are too busy i understand, and i'm thankful anyway for you creating this project. 🙏

**Reply by seerge** [reply]
You just replace 60 with whatever you need here
https://github.com/seerge/g-helper/blob/1efda38ccf9971f593f9ca1219ced0744a68f16e/app/Display/ScreenControl.cs#L18
